components/sync/service/brave_sync_service_impl_unittest.cc

```diff
@@ -708,4 +712,42 @@ TEST_F(BraveSyncServiceImplTest, NoLeaveDetailsWhenInitializeIOS) {
   EXPECT_FALSE(brave_sync_prefs()->GetLeaveChainDetails().empty());
 }
 
+TEST_F(BraveSyncServiceImplTest, OnAccountsCookieDeletedByUserAction) {
+  SyncTransportDataPrefs::RegisterProfilePrefs(pref_service()->registry());
+
+  SyncTransportDataPrefs sync_transport_data_prefs(
+      pref_service(), signin::GaiaIdHash::FromGaiaId("user_gaia_id"));
+
+  sync_transport_data_prefs.SetCacheGuid(kCacheGuid);
+
+  CreateSyncService();
+
+  brave_sync_service_impl()->OnAccountsCookieDeletedByUserAction();
+
+  EXPECT_EQ(sync_transport_data_prefs.GetCacheGuid(), kCacheGuid);
+}
+
+TEST_F(BraveSyncServiceImplTest, OnAccountsInCookieUpdated) {
+  SyncTransportDataPrefs::RegisterProfilePrefs(pref_service()->registry());
+
+  SyncTransportDataPrefs sync_transport_data_prefs(
+      pref_service(), signin::GaiaIdHash::FromGaiaId("user_gaia_id"));
+
+  sync_transport_data_prefs.SetCacheGuid(kCacheGuid);
+
+  CreateSyncService();
+
+  brave_sync_service_impl()->OnAccountsInCookieUpdated(
+      signin::AccountsInCookieJarInfo(), GoogleServiceAuthError());
+
+  EXPECT_EQ(sync_transport_data_prefs.GetCacheGuid(), kCacheGuid);
+}
+
+// No test for SyncServiceImplBrave::OnPrimaryAccountChanged
```

Comment:
We can still test it even though we know it is not going to change anything now even without override. 
It's also a measure to make sure no matter how the upstream function changes in the future, the cache guid won't be affected.

Reply:
thanks, addressed at 
https://github.com/brave/brave-core/pull/25971/commits/262e95654eaacbfd7425e211d587c82bbacd290b